scene/2d/parallax_2d.cpp

```diff
@@ -0,0 +1,293 @@
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "parallax_2d.h"
+
+#include "core/config/project_settings.h"
+
+void Parallax2D::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE: {
+			group_name = "__cameras_" + itos(get_viewport_rid().get_id());
+			add_to_group(group_name);
+			_update_repeat();
+			_update_scroll();
+		} break;
+
+		case NOTIFICATION_READY: {
+			_update_process();
+		} break;
+
+		case NOTIFICATION_INTERNAL_PROCESS: {
+			autoscroll_offset += autoscroll * get_process_delta_time();
+			autoscroll_offset = autoscroll_offset.posmodv(repeat_size);
+
+			_update_scroll();
+		} break;
+
+		case NOTIFICATION_EXIT_TREE: {
+			remove_from_group(group_name);
+		} break;
+	}
+}
+
+#ifdef TOOLS_ENABLED
+void Parallax2D::_edit_set_position(const Point2 &p_position) {
+	set_scroll_offset(p_position);
+}
+#endif // TOOLS_ENABLED
+
+void Parallax2D::_validate_property(PropertyInfo &p_property) const {
+	if (p_property.name == "position") {
+		p_property.usage = PROPERTY_USAGE_NO_EDITOR;
```

Comment:
NO_EDITOR makes the property hidden from the inspector, but still saved in the scene. Is the value relevant for serialization?

Reply:
I'm not sure if it makes any difference, since if it's not saved and starts at `0,0` the very first thing that will happen is that the position will update to whatever the offset is... so I don't think one is more beneficial than the other. Is there a reason you would prefer it to not save with the offset?

Reply:
If upon load the position is set from the offset anyway, then saving it is redundant. It just makes unnecessary entry for the node.

Reply:
I don't have any preference between the two, so I'll go with yours!